internal/backends/sqlite/metadata/registry.go

```diff
@@ -346,11 +364,56 @@ func (r *Registry) CollectionDrop(ctx context.Context, dbName, collectionName st
 
 // CollectionRename renames a collection in the database.
 //
+// The collection name is update, but original table name is kept.
+//
 // Returned boolean value indicates whether the collection was renamed.
 // If database or collection did not exist, (false, nil) is returned.
 func (r *Registry) CollectionRename(ctx context.Context, dbName, oldCollectionName, newCollectionName string) (bool, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/2760
-	panic("not implemented")
+	defer observability.FuncCall(ctx)()
+
+	db := r.p.GetExisting(ctx, dbName)
+	if db == nil {
+		return false, nil
+	}
+
+	r.rw.Lock()
+	defer r.rw.Unlock()
+
+	colls := r.colls[dbName]
+	if colls == nil {
+		return false, nil
+	}
+
+	cOld := colls[oldCollectionName]
+	if cOld == nil {
+		return false, nil
+	}
+
+	q := fmt.Sprintf(`UPDATE %q SET name = ? WHERE table_name = ?`, metadataTableName)
+
+	res, err := db.ExecContext(ctx, q, newCollectionName, cOld.TableName)
+	if err != nil {
+		return false, lazyerrors.Error(err)
+	}
+
+	ra, err := res.RowsAffected()
```

Comment:
I don't think there is a need to check `res` – any value other than 1 means that cache and data in SQLite are not in-sync, and that should not be possible

Reply:
Indeed, in theory it shouldn't be possible. Remove the check to simplify it :+1: